projects/core/src/routing/event/routing-event.builder.ts

```diff
@@ -0,0 +1,164 @@
+import { Injectable } from '@angular/core';
+import { ofType } from '@ngrx/effects';
+import { RouterNavigatedAction, ROUTER_NAVIGATED } from '@ngrx/router-store';
+import { ActionsSubject } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { filter, map, switchMap, withLatestFrom } from 'rxjs/operators';
+import { EventService } from '../../event/event.service';
+import { ProductSearchService } from '../../product/facade/product-search.service';
+import { ProductService } from '../../product/facade/product.service';
+import { createFrom } from '../../util/create-from';
+import { SemanticPathService } from '../configurable-routes';
+import { PageContext } from '../models/page-context.model';
+import { ActivatedRouterStateSnapshot } from '../store/routing-state';
+import {
+  CartPageVisited,
+  CategoryPageVisited,
+  HomePageVisited,
+  SearchPageVisited,
+  OrderConfirmationPageVisited,
+  PageVisited,
+  ProductDetailsPageVisited,
+} from './routing.events';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class RoutingEventBuilder {
+  constructor(
+    protected actions: ActionsSubject,
+    protected eventService: EventService,
+    protected productService: ProductService,
+    protected productSearchService: ProductSearchService,
+    protected semanticPathService: SemanticPathService
+  ) {
+    this.register();
+  }
+
+  protected register(): void {
+    this.eventService.register(
+      SearchPageVisited,
```

Comment:
Medium: For consistency, it would be good to end events' names with the word `Event` (i.e. `SearchPageVisitedEvent`). Same for other events.

Reply:
It appears this has been moved to another PR